File: test-code/src/main/java/com/example/testcode/member/MemberApi.java
package com.example.testcode.member;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/members")
@RequiredArgsConstructor
public class MemberApi {

  private final MemberRepository memberRepository;

  @PostMapping
  public Member create(@RequestBody MemberSingUpRequest dto) {
    return memberRepository.save(new Member(dto.getEmail(), dto.getName()));
  }

  @GetMapping
  public Page<Member> getOrderAll(Pageable pageable) {
    return memberRepository.findAll(pageable);
  }

}

File: spring-kotlin/src/main/kotlin/com/example/springkotlin/member/api/MemberApi.kt
package com.example.springkotlin.member.api

import com.example.springkotlin.member.dao.MemberRepository
import com.example.springkotlin.member.domain.Member
import com.example.springkotlin.member.dto.MemberSignUpRequest
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.*


@RestController
@RequestMapping("/members")
class MemberApi constructor(
        private var memberRepository: MemberRepository) {

    @PostMapping
    fun createMember(@RequestBody dto: MemberSignUpRequest): Member {
        return memberRepository.save(dto.toEntity())
    }

    @GetMapping
    fun getMembers(page: Pageable): Page<Member> {
        return memberRepository.findAll(page)
    }

}